Add info and disabled state for tasks

renatho opened the pull request:
Part of https://github.com/Automattic/sensei-pro/issues/2540

## Proposed Changes

* It adds info prop for tasks. With this, we can add an info icon with a tooltip.
* It adds a disabled state for tasks, so the task is not clickable.

## Testing Instructions
<!--
Add as many details as possible to help others reproduce the issue and test the changes.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

See https://github.com/Automattic/sensei-pro/pull/2565

## Pre-Merge Checklist
<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
- [x] PR title and description contain sufficient detail and accurately describe the changes
- [x] Acceptance criteria is met
- [x] Decisions are publicly documented
- [x] Adheres to coding standards ([PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/), [JavaScript](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/), [CSS](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), [HTML](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/))
- [x] All strings are translatable (without concatenation, handles plurals)
- [x] Follows our naming conventions (P6rkRX-4oA-p2)
- [ ] Hooks (p6rkRX-1uS-p2) and functions are documented
- [x] New UIs are responsive and use a [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/)
- [x] New UIs match the designs
- [ ] Different user privileges (admin, teacher, subscriber) are tested as appropriate
- [ ] Legacy courses (course without blocks) are tested
- [ ] Code is tested on the minimum supported PHP and WordPress versions
- [ ] User interface changes have been tested on the latest versions of Chrome, Firefox and Safari
- [ ] "Needs Documentation" label is added if this change requires updates to documentation
- [ ] Known issues are created as new GitHub issues

renatho committed:
Add wp-components style to the Sensei Home

So tooltip can be used with styles.

renatho committed:
Add support for disabled state and info text in the texts

renatho committed:
Add tests

renatho commented on includes/admin/class-sensei-home.php:
```diff
@@ -143,7 +143,7 @@ public function enqueue_admin_assets() {
 		if ( self::SCREEN_ID === $screen->id ) {
 			Sensei()->assets->enqueue( 'sensei-home', 'home/index.js', [], true );
-			Sensei()->assets->enqueue( 'sensei-home-style', 'home/home.css', [ 'sensei-wp-components' ] );
+			Sensei()->assets->enqueue( 'sensei-home-style', 'home/home.css', [ 'sensei-wp-components', 'wp-components' ] );
```
`wp-components` to load the tooltip styles in Sensei Home.

Imran92 committed:
Merge branch 'trunk' into add/info-and-disabled-state-for-tasks

Imran92 reviewed (dismissed):
LGTM! 👍 I just have a small suggestion to update the documentation here to inform the developers that it's possible to add an info icon using the hook https://github.com/Automattic/sensei/blob/f6a76551a5e6b154e82426c89dc31563b77f504d/includes/admin/home/tasks/class-sensei-home-tasks-provider.php#L89-L105

renatho committed:
Merge branch 'trunk' of github.com:Automattic/sensei into add/info-and-disabled-state-for-tasks

renatho committed:
Only display external icon if task is active

renatho committed:
Fix icon position

renatho committed:
Document new task properties

renatho commented:
Thank you for the review, @Imran92!

> I just have a small suggestion to update the documentation here to inform the developers that it's possible to add an info icon using the hook

I didn't see the suggestion, but I think I got what you meant and I added this commit: https://github.com/Automattic/sensei/pull/7564/commits/da1077613e456e45de72728a1bc692fa3d144f0e

Besides that, after updating the branch with the changes on `trunk`, I fixed a case for when the task is done or disabled. In this case, I removed the external icon. I think it's correct, right? https://github.com/Automattic/sensei/pull/7564/commits/fe966b73a6c5358ab28c66d1b65183c12470acd4

Imran92 reviewed (approved):
> I didn't see the suggestion, but I think I got what you meant and I added this commit: https://github.com/Automattic/sensei/commit/da1077613e456e45de72728a1bc692fa3d144f0e
> 
> Besides that, after updating the branch with the changes on trunk, I fixed a case for when the task is done or disabled. In this case, I removed the external icon. I think it's correct, right? https://github.com/Automattic/sensei/commit/fe966b73a6c5358ab28c66d1b65183c12470acd4

Yap, all looks good now, LGTM! 👍
